torchaudio/functional.py

```diff
@@ -419,12 +419,80 @@ def DB_to_amplitude(
     return ref * torch.pow(torch.pow(10.0, 0.1 * x), power)
 
 
+def _hz_to_mel(freq: Tensor, mel_scale: str = "htk") -> Tensor:
+    r"""Convert Hz to Mels
+
+    Args:
+        freqs (Tensor): Frequencies in Hz
+        mel_scale (str, optional): Scale to use: ``htk`` or ``slaney``. (Default: ``htk``)
+
+    Returns:
+        mels (Tensor): Input frequencies in Mels
+    """
+
+    if mel_scale == "htk":
+        return 2595.0 * torch.log10(torch.tensor(1.0 + (freq / 700.0), dtype=torch.float64))
+    elif mel_scale == "slaney":
+
+        # Fill in the linear part
+        f_min = 0.0
+        f_sp = 200.0 / 3
+
+        mels = (freq - f_min) / f_sp
+
+        # Fill in the log-scale part
+        min_log_hz = 1000.0
+        min_log_mel = (min_log_hz - f_min) / f_sp
+        logstep = math.log(6.4) / 27.0
+
+        log_t = freq >= min_log_hz
+        mels[log_t] = min_log_mel + torch.log(torch.tensor(freq[log_t] / min_log_hz, dtype=torch.float64)) / logstep
+
+        return mels
+    else:
+        raise ValueError('mel_scale should be one of "htk" or "slaney".')
+
+
+def _mel_to_hz(mels: Tensor, mel_scale: str = "htk") -> Tensor:
+    """Convert mel bin numbers to frequencies
+
+    Args:
+        mels (Tensor): Mel frequencies
+        mel_scale (str, optional): Scale to use: ``htk`` or ``slaney``. (Default: ``htk``)
+
+    Returns:
+        freqs (Tensor): Mels converted in Hz
+    """
+
+    if mel_scale == "htk":
+        return 700.0 * (10.0**(mels / 2595.0) - 1.0)
+    elif mel_scale == "slaney":
+
+        # Fill in the linear scale
+        f_min = 0.0
+        f_sp = 200.0 / 3
+        freqs = f_min + f_sp * mels
+
+        # And now the nonlinear scale
+        min_log_hz = 1000.0
+        min_log_mel = (min_log_hz - f_min) / f_sp
+        logstep = math.log(6.4) / 27.0
+
+        log_t = (mels >= min_log_mel)
+        freqs[log_t] = min_log_hz * torch.exp(logstep * (mels[log_t] - min_log_mel))
+
+        return freqs
+    else:
+        raise ValueError('mel_scale should be one of "htk" or "slaney".')
+
+
 def create_fb_matrix(
         n_freqs: int,
         f_min: float,
         f_max: float,
         n_mels: int,
-        sample_rate: int
+        sample_rate: int,
+        mel_scale: str = "htk",
 ) -> Tensor:
```

Comment:
It seems like the new implementation returns float64 Tensor whereas the original implementation returns float32. If so, this is BC breaking.

Either case, I think it's good idea to document what dtype the resulting tensor is.

Reply:
Good point. I'll cast to `float32` at the end of the function. We can always add a `dtype` parameter to `create_fb_matrix` if other types are needed later.

The dtype returned should be documented in all functions indeed.

Reply:
Conversions like that should incur a fully copy. I'd double check what the performance implications of using float64 are. We could also set expectations and let the user pass float64 Tensors if they want higher precision.

Reply:
I agree with the idea, but I would leave that discussion as a separate PR, since this may affect the outcome of #611.